### ISSUE TITLE
Saved scores do not count each note individually in BM mode

### ISSUE THREAD

**Issue by ca25nada** [stepmania/stepmania#989]
Note: Not sure if this applies to every gamemode where CountNotesSeparately() is true.

Saved scores in BM mode still treats chords (e.g. jumps would be a 2-note chord) as a single judgment whereas the judgments occur separately for each note inside a chord during gameplay.

Here's an example with a bms file containing 3000 notes / 1053 taps

2729 W1 , 271 W2 from counting within the theme. 
![stepmania 2016-01-14 22-50-50-27](https://cloud.githubusercontent.com/assets/4557043/12346642/7109e2ec-bb13-11e5-9f46-5cfeacfbd9db.png)

977 W1 and 76 W2 when grabbing values from either PlayerStageStats:GetTapNoteScores(tns) or HighScore:GetTapNoteScore(tns)
![stepmania 2016-01-14 22-50-55-36](https://cloud.githubusercontent.com/assets/4557043/12346657/9f66585a-bb13-11e5-9232-22a80fe4fb46.png)

**Comment by parashep** [comment]
I believe I can confirm this for the pop'n game mode as well? it looks like calling GetTapNoteScores() during gameplay returns the correct value but once the song ends and the screen changes the value returned is no longer correct. I will take a deeper look into solving this sometime soon

**Comment by kyzentun** [comment]
Probably some mistake in NoteDataWithScoring.cpp.  Need to make it check the chord stuff.

**Comment by nixtrix** [comment]
Stepmania's built-in scoring system counts double notes (jumps etc.) as one. This is a throwback from DDR/ITG/pump/etc.'s scoring systems, which all count jumps as one note (except for combo - hence max 300's 555 notes but max combo of 573). To fix this you can either edit the source code, or write your own scoring system in Lua BGAnimations (which I'm planning on doing for BeatFreeX).

**Comment by parashep** [comment]
right, but is that not what the `m_bCountNotesSeparately` Game setting is for? (see GameManager.cpp)
I would imagine that game modes that set that as `true` (pop'n, IIDX) should store a proper value, especially since the value given during gameplay is correct.

**Comment by nixtrix** [comment]
You'd think so, wouldn't you? I edited the source code to enable m_bCountNotesSeparately for the IIDX game mode, but it appears to be extremely precise in its naming in that while it _counts_ notes separately, it doesn't actually _score_ them separately. So yeah, I believe it shows the correct number of notes on the song selection screen (not certain about this one because my theme doesn't show note counts just yet) but scoring still works as it would in DDR etc.

**Comment by Kaiveran** [comment]
Should probably reference #605 here. When options to turn off chord cohesion are available, they should be off for these modes.

**Comment by Flameshadowxeroshin** [comment]
(i don't like chord cohesion, it's a made up word)
m_bCountNotesSeparately effectively doesn't work, unless something has changed since this issue was made
